src/main/kotlin/app/cash/licensee/pluginExtension.kt

```diff
@@ -245,8 +247,19 @@ internal abstract class IgnoredCoordinate : Named {
   abstract val ignoredDatas: MapProperty<String, IgnoredData>
 }
 
-internal abstract class MutableLicenseeExtension : LicenseeExtension {
-  internal abstract val allowedIdentifiers: SetProperty<String>
+internal abstract class SpdxId(private val spdxId: String) : Named {
+  override fun getName(): String = spdxId
+  init {
+    requireNotNull(SpdxLicenses.embedded.findByIdentifier(spdxId)) {
+      "$name is not a valid SPDX id."
+    }
```

Comment:
It seems like this could have been put in the old `allow` before adding to the set to accomplish the same thing. Is there another advantage of all the other changes that I'm missing. Is it secretly working towards #313?

Reply:
> Is it secretly working towards https://github.com/cashapp/licensee/issues/313?

Yes, this is the only reason. As you can see in the commits I added the provider overload for `allow` but realized all functions could benefit of `Provider`s so I removed it here.
We could also add the provider overload back and add more in #313.

Reply:
I would prefer them separate, but we can leave the infrastructure in place here.

Reply:
BTW I don't plan to implement #313 today and it needs some design first, in case you want to publish another release.

Reply:
Will do